### ISSUE TITLE
Thanks for putting this together

### ISSUE THREAD

**Issue by Mogball** [dsharlet/mojo_comments#1]
I just want to say thanks for the detailed breakdown of a comparison between Mojo and C++ (I assume you're using clang?). I'll take an AI to investigate why Mojo is generating worse code. For one, we don't use the same LLVM optimization pipeline as clang does, but also the Mojo end of the compiler probably just needs some work.

Cheers!

**Comment by dsharlet** [comment]
Yes, I'm using clang, but you can get very similar results with GCC if using -O3 (I use -O2 with clang). Glad to hear someone found this and thought it was useful :)
